### PR TITLE
Register start and end time

### PR DESCRIPTION
**Pull request by InzeNL**
Partial implemention for #3 
This will not close this issue yet.

This pull-request adds the following functionality:
- Recording start and end time
- Fixed-time reservation. This means, when someone cancels the reservation before you, your reservation doesn't move up
- Automatically inserts the reservation in the first available timeslot

This pull-request does not yet add the ability to select a specific start-time. This will be implemented separately, likely after the refactor for #9

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -86,20 +115,57 @@ def GitPull():
...
+            startTime = current.endTime
+        # If there is no current reservation, and the new reservation will end before the first next reservation starts
+        elif current is None and dateTimeNow + minutes <= queue[0].startTime:
+            startTime = dateTimeNow
+        # If there is no current reservation, and there is only one reservation in the queue
+        elif current is None and len(queue) == 1:
+            startTime = queue[0].endTime
+        # If the new reservation would not fit between the current and the first reservation
+        else:
+            currentItem = queue[0]
```
Comment (DavidBergevoet) [inline app.py:139]: Make current item `current` and start loop at 1. If the current has end time 12:12 and queue[0] starts at 12:20, a reservation of 5 minutes will now just pop to the end of the queue. By setting the currentItem to current and start the queue at 0 it should set the right start time in the while loop

File: app.py [hunk 2]
```diff
@@ -157,7 +223,7 @@ def version_request():
 if __name__ == '__main__':
     UpdateQueueFromFile()
     scheduler = BackgroundScheduler()
-    scheduler.add_job(func=UpdateCurrentTimer, trigger="interval", seconds=60)
+    scheduler.add_job(func=UpdateCurrentTimer, trigger="interval", seconds=1)
```
Comment (DavidBergevoet) [inline app.py:226]: Maybe its better to set this to 10 seconds or so, otherwise there are a lot of file writes

**Review by DavidBergevoet — CHANGES_REQUESTED** [review]

File: app.py (hunk 1 above)
Comment (InzeNL) [reply app.py]: 1e3cf13091c657d8b7dbfedd507644856cc27a14

File: app.py (hunk 2 above)
Comment (InzeNL) [reply app.py:237]: Instead, I opted to only write to the file when changes were applied
e31c45ec6961759e3c6d873254ffbe6c2acad14b

**Review by DavidBergevoet — APPROVED** [review]

**Pull request merged**